### PR TITLE
Add hidden content & check against it before adding pivotal story details

### PR DESCRIPTION
**Pull request by rheaditi**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -179,48 +180,65 @@ export const getStoryIcon = (storyType: string): string => {
  * @returns string
  */
 export const getPrDescription = (body: string = '', story: StoryResponse): string => {
+  if (!shouldPrependPivotalStats(body)) {
```
Comment (rajanand02) [inline src/utils.ts]: if we do it outside this method then we can prevent the API call itself, right? with this we are triggering a call always

Comment (rajanand02) [reply src/utils.ts]: and the method is called before it is defined. we should move the method before `getPrDescription`

Comment (rajanand02) [reply src/utils.ts]: do this check in `main.ts` so that we can avoid calling `updatePrDetails`

Comment (rheaditi) [reply src/utils.ts]: > if we do it outside this method then we can prevent the API call itself, right? with this we are triggering a call always

which API call? we have to get the PR details to know if it should be added..

Comment (rheaditi) [reply src/utils.ts]: ohh got it

Comment (rajanand02) [reply src/utils.ts]: PR details are already part of the triggered action. we aren't making any `GET` call to get the body. 

```js
 if(shouldPrependPivotalStats(prBody)) {
      const prData: PullsUpdateParams = {
        owner: organization.login,
        repo,
        pull_number: prNumber,
        body: getPrDescription(prBody, story),
      };
      updatePrDetails(client, prData);
}
```

File: src/utils.ts [hunk 2]
```diff
@@ -172,55 +173,74 @@ export const getStoryIcon = (storyType: string): string => {
   }
 };
 
+export const shouldPrependPivotalStats = (
```
Comment (rajanand02) [inline src/utils.ts]: test and jsdoc :nerd_face:

File: src/main.ts [hunk 3]
```diff
@@ -77,13 +78,15 @@ async function run() {
...
-      };
-      updatePrDetails(client, prData);
+      if (shouldUpdatePRDescription(prBody)) {
+        const prData: PullsUpdateParams = {
+          owner: organization.login,
+          repo,
+          pull_number: prNumber,
+          body: getPrDescription(prBody, story),
+        };
+        await updatePrDetails(client, prData);
```
Comment (rajanand02) [inline src/main.ts]: do we need an `await` if we already using an `await` inside the method?

Comment (rheaditi) [reply src/main.ts]: IIRC async functions return promises right? else the code after it will run before the internal awaits..

Comment (rajanand02) [reply src/main.ts]: yes it does return promise. Can you please add to the other async call as well where we addLabel

Comment (rajanand02) [reply src/main.ts]: we were just calling the methods and not waiting for the result. and btw if we are not utilizing the response do we need an await?

Comment (rajanand02) [reply src/main.ts]: I agree that it returns a promise. this is like fire and forget call. :thinking:

File: src/utils.ts (hunk 2 above)
Comment (rheaditi) [reply src/utils.ts]: done

File: src/main.ts (hunk 3 above)
Comment (rheaditi) [reply src/main.ts]: added await for `addLabels` as well

File: lib/index.js [hunk 4]
```diff
@@ -4077,6 +4080,16 @@ exports.getStoryIcon = (storyType) => {
...
 };
+/**
+ * Returns true if the body contains the hidden marker. Used to avoid adding
+ * pivotal story details to the PR multiple times.
+ *
+ * @example shouldUpdatePRDescription('--\nadded_by_pr_lint\n') -> true
+ * @example shouldUpdatePRDescription('# some description') -> false
+ */
+exports.shouldUpdatePRDescription = (
+/** The PR description/body as a string. */
```
Comment (rajanand02) [inline lib/index.js]: no `@params {string} body`?

**Pull request merged**